File: AQ_LIB/resources/api/RExamples/CurveDiscFactorsWithSpread/M_DiscountFactorsWithSpread.R
dyn.load("MLIB_CLIENT_API")
source("MLIB_CLIENT_API.R")
getLoadedDLLs()
setUpMLIB("","")
meUtilityVersion()


oisFilename <- c("C:/Temp/EUR_OIS_CURVE@30.JSON")
stdFilename <- c("C:/Temp/EUR_SWAP_3M_CURVE@18.JSON")

meLWOLoad( oisFilename )
meLWOLoad( stdFilename )

#Set Payment Date Vectors
payDates <- c( "20190101", "20190401" )

#Get Discount Factor(s) - Single or Multiple Dates Accepted
meLWOCurveDiscountFactorsWithSpread(payDates,"EURYC","EUR3ML", 0.0) 

# 1bp spread
meLWOCurveDiscountFactorsWithSpread(payDates,"EURYC","EUR3ML", 0.0001)

File: AQ_LIB/resources/api/PythonExamples/CurvesDiscFactorsWithSpread/M_DiscountFactorsWithSpread.py
#Import the MLIB CLIENT API
import MLIB_CLIENT_API as m

#Load Calendars and Config Files
print ( m.setUpMLIB("","") )

#Get the MLIB Version
print ( m.meUtilityVersion() )

print ( m.meLWOLoad("C:\Temp\EUR_OIS_CURVE@30.JSON") )
print ( m.meLWOLoad("C:\Temp\EUR_SWAP_3M_CURVE@18.JSON") )

#Set Payment Date Vectors
payDates = ["20190101", "20190401"]

#Get Discount Factor(s) - Single or Multiple Dates Accepted
print ( "Discount Factor" )
print ( m.meLWOCurveDiscountFactorsWithSpread(payDates,"EURYC","EUR3ML", 0.0) )

# 1bp spread
print ( m.meLWOCurveDiscountFactorsWithSpread(payDates,"EURYC","EUR3ML", 0.0001) )

